ngraph/core/src/op/batch_to_space.cpp

```diff
@@ -134,3 +140,115 @@ bool ngraph::op::v1::BatchToSpace::visit_attributes(ngraph::AttributeVisitor& vi
 {
     return true;
 }
+
+bool ngraph::op::v1::BatchToSpace::evaluate(const HostTensorVector& outputs,
+                                            const HostTensorVector& inputs) const
```

Comment:
I don't see any reasons to store this reference in a separate file in this case, only for unification maybe. But in general, since it's not a templated reference it can't be reused anywhere (at least for now) e.g. in interpreter as it's done sometimes as a fallback for data types, which not instantiated in evaluate.

As far as I remember, I discussed this with @ilyachur. What are your thoughts, about adding reference implementation directly to evaluate in such cases?

Reply:
That's fine if you ask me. Just wanted to know the reason.

Reply:
@mikhail-treskin We don't discuss it.
We discuss only a common recommendations for the reference implementation development.
If you have a reference implementation without a template code, you can use add it to evaluate method (just because each supported data type doesn't affect the binary footprint).

But as I understand right @mateusztabaka has another question. At the current moment we create reference implementations for operations in the separate library and only after that use these implementations in the evaluate methods. This approach allow to split operation and reference implementations.

I think we can create a reference implementation for BatchToSpace and after that call it here.

Reply:
@mikhail-treskin We don't discuss it.
We discuss only a common recommendations for the reference implementation development.
If you have a reference implementation without a template code, you can use add it to evaluate method (just because each supported data type doesn't affect the binary footprint).

But as I understand right @mateusztabaka has another question. At the current moment we create reference implementations for operations in the separate library and only after that use these implementations in the evaluate methods. This approach allow to split operation and reference implementations.

I think we can create a reference implementation for BatchToSpace and after that call it here.

Reply:
Ok, I see pros and cons for both approaches, but let's stick to the approach which is most common - I'll move the implementation to  ngraph/core/reference

Reply:
Decided to move to `ngraph/core/reference` in separate PR